Modules/Loadable/Markups/MRML/vtkSlicerDijkstraGraphGeodesicPath.h

```diff
@@ -0,0 +1,87 @@
+/*==============================================================================
+
+  Copyright (c) Laboratory for Percutaneous Surgery (PerkLab)
+  Queen's University, Kingston, ON, Canada. All Rights Reserved.
+
+  See COPYRIGHT.txt
+  or http://www.slicer.org/copyright/copyright.txt for details.
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  This file was originally developed by Kyle Sunderland, PerkLab, Queen's University
+  and was supported through CANARIE's Research Software Program, Cancer
+  Care Ontario, OpenAnatomy, and Brigham and Women�s Hospital through NIH grant R01MH112748.
+
+==============================================================================*/
+
+#ifndef __vtkSlicerDijkstraGraphGeodesicPath_h
+#define __vtkSlicerDijkstraGraphGeodesicPath_h
+
+// VTK includes
+#include <vtkDijkstraGraphGeodesicPath.h>
+
+// export
+#include "vtkSlicerMarkupsModuleMRMLExport.h"
+
+//
+// Set built-in type.  Creates member Set"name"() (e.g., SetVisibility());
+//
+#define vtkSetAdjacencyMacro(name,type) \
```

Comment:
Why do we need this, why here, and with a macro?

Reply:
In the base class, the adjacency is only recalculated if the input is modified. I added the RecalculateAdjacency varaible that functions could set to indicate that edge calculation parameters have changed, and that the Adjacency should be recomputed.

I added this macro (and created overridden methods for BuildAdjacency and RequestData) to set/use this variable, rather than copy-pasting the SetX macro for each function that needed the adjacency to be rebuilt.

Reply:
OK, I understand now. Set macros are somewhat opaque but they are so simple that they are acceptable. However, in this case, it makes things a bit too complicated. Simpler solutions could be to just repeating those few lines in simple methods; or do things like storing of the CostFunction and UseScalarWeights values that were used when computing adjacency last time, or store last time adjacency related input values changed in a timestamp - and do not recompute adjacency if not needed.